compiler/next/lib/resolution/resolution-queries.cpp

```diff
@@ -2272,11 +2291,37 @@ CallResolutionResult resolveFnCall(Context* context,
                                    const PoiScope* inPoiScope) {
   // search for candidates at each POI until we have found a candidate
   std::vector<const TypedFnSignature*> candidates;
-  size_t firstPoiCandidate = 0;
   std::unordered_set<const Scope*> visited;
   PoiInfo poiInfo;
 
-  // first, look for candidates without using POI.
+  // If call is a method, search receiver definition scope for methods.
+  if (ci.isMethod()) {
+    assert(ci.numActuals() >= 1);
+
+    // Receiver should be first actual of method call.
+    const auto& receiver = ci.actuals(0);
+
+    if (auto compType = receiver.type().type()->getCompositeType()) {
+      auto candidateScope = scopeForId(context, compType->id());
+      assert(candidateScope);
+
+      // compute the potential functions that it could resolve to
+      auto v = lookupCalledExpr(context, inScope, call,
```

Comment:
AFAIK these next three calls are redundant with what happens starting on line 2328. I recommend you just compute `candidateScope` in the `if method` condition and pass `nullptr` for it if it is not a method. That should allow fewer lines of code and less redundancy.